GoogleForCreators/web-stories-wp
# Block: Fix carousel width & whitespace issues

## Context



This PR fixes two issues:

First, the Glider track width calculation was quite off, causing white space to appear at the beginning of the carousel. Noticeable mostly on Firefox, but also on Chrome.


## Summary



Fixes a regression with the carousel layout in v1.26.0.

Also avoids unnecessary whitespace at the beginning of the carousel

## Relevant Technical Choices



## To-do



## User-facing changes



**Whitespace issue**

See: https://www.youtube.com/watch?v=dCpdV3PurC0

**Width issue**

Before:

<img width="1074" alt="Screenshot 2022-10-31 at 10 02 25" src="https://user-images.githubusercontent.com/841956/198971998-5d7d0a8b-dd4b-46dc-bfe5-9c58071d3fa2.png">

After:

<img width="963" alt="Screenshot 2022-10-31 at 10 05 42" src="https://user-images.githubusercontent.com/841956/198972014-35d4daf8-13d1-4d7c-afc7-d8de8d580c38.png">

## Testing Instructions






This PR can be tested by following these steps:

1. Add a new Web Stories block to a post, make sure there are plenty of stories in it.
2. Use the grid carousel and circle carousel layout options
3. View in Firefox w/ small viewport (+ large ones to ensure backwards compat)
4. Previous gap should be fixed
5. Carousel should be displayed properly in full width

For completeness sake, also test other layout options to see if there are no regressions.

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #12563
Fixes #12170
